lib/meandro/rules/unused_macros.ex

```diff
@@ -0,0 +1,116 @@
+defmodule Meandro.Rule.UnusedMacros do
+  @moduledoc """
+  Finds macros that are not used.
+  """
+
+  @behaviour Meandro.Rule
+
+  @impl Meandro.Rule
+  def analyze(files_and_asts, _options) do
+    for {file, module_asts} <- files_and_asts,
+        {_module_name, ast} <- module_asts,
+        result <- analyze_module(file, ast, files_and_asts) do
+      result
+    end
+    |> List.flatten()
+  end
+
+  @impl Meandro.Rule
+  def is_ignored?({module, macro}, {module, macro}) do
+    true
+  end
+
+  def is_ignored?(_pattern, _ignore_spec) do
+    false
+  end
+
+  defp analyze_module(file, ast, files_and_asts) do
+    module_aliases = Meandro.Util.module_aliases(ast)
+    macros = macros(ast, module_aliases)
+
+    for macro_info <- macros do
+      unused = is_unused?(macro_info, files_and_asts)
+      macro = macro_info |> Map.get(:name)
```

Comment:
Can we have multiple macros with the same name? Like…
```elixir
defmacro my_macro do
…
end
defmacro my_macro(x, second_param: y, thrid_param: z) do
…
end
```

Reply:
Yep, now it takes that into account